DOC add example to reorder levels

trevorkask opened the pull request:
Response to changes suggested on my previous pull request

- [x] closes #42124
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

trevorkask committed:
Update frame.py

Response to changes suggested on my previous pull request

pep8speaks commented:
Hello @trevorkask! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`pandas/core/frame.py`](https://github.com/pandas-dev/pandas/blob/61bd74917f46b5d83d2f53286a1748429a83d1ac/pandas/core/frame.py):

> [Line 6793:8](https://github.com/pandas-dev/pandas/blob/61bd74917f46b5d83d2f53286a1748429a83d1ac/pandas/core/frame.py#L6793): [E901](https://duckduckgo.com/?q=pep8%20E901) IndentationError: unindent does not match any outer indentation level

MarcoGorelli reviewed (changes requested):
Have you tried building the docs to check this looks right? 

https://pandas.pydata.org/pandas-docs/stable/development/contributing_documentation.html

attack68 commented:
Are you going to close #42667 or this? You should really continue that PR and close this one.

trevorkask committed:
Update frame.py

Fixed the indentation issue

MarcoGorelli commented on pandas/core/frame.py:
```diff
@@ -6790,7 +6790,33 @@ def swaplevel(self, i: Axis = -2, j: Axis = -1, axis: Axis = 0) -> DataFrame:
             result.columns = result.columns.swaplevel(i, j)
         return result
-
+        """
+        Examples
+        --------
```
This won't render, it needs to be inside the docstring

MarcoGorelli commented on pandas/core/frame.py:
```diff
@@ -6790,7 +6790,33 @@ def swaplevel(self, i: Axis = -2, j: Axis = -1, axis: Axis = 0) -> DataFrame:
+         'species':['Humans','Dogs','Snakes','Iguanas','Frogs','Aquatic Salamanders']
+         }
+        >>> df=pd.DataFrame(data,columns=['class','diet','species'])
+        >>> df.set_index(['class','diet'],drop=True,inplace=True)
```
can you write this without using `inplace=True` please?

MarcoGorelli reviewed (changes requested):
Can you try building the docs for this function locally to check it renders fine? (hint: at the moment it won't, this'll help you tell whether the PR is ready)

jreback commented:
use this one

trevorkask committed:
Update frame.py

Built the example inside the docstring and removed the unnecessary inplace=true

MarcoGorelli reviewed (changes requested):
Hi @trevorkask - the example looks good, there's just some stylistic errors

Can you fix them up? See https://pandas.pydata.org/pandas-docs/stable/development/contributing_codebase.html?highlight=pre%20commit#pre-commit

MarcoGorelli committed:
Merge remote-tracking branch 'upstream/master' into pr/trevorkask-patch-2

MarcoGorelli committed:
fixup

MarcoGorelli committed:
Merge remote-tracking branch 'upstream/master' into pr/trevorkask-patch-2

MarcoGorelli reviewed (approved):
![](https://i.fluffy.cc/Xr1jnB3r6dWrfwqPwNdrpnF9zW1lVp0F.gif)

MarcoGorelli commented:
Thanks @trevorkask
